src/armor_layers.cpp

```diff
@@ -112,387 +106,481 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
-    props.push_back( string_format( "<color_c_green>[%s]</color>", _( "Properties" ) ) );
-    props.push_back( name_and_value( space + _( "Coverage:" ),
-                                     string_format( "%3d", worn_item.get_coverage() ), width ) );
-    props.push_back( name_and_value( space + _( "Encumbrance:" ),
-                                     string_format( "%3d", worn_item.get_encumber( c ) ),
-                                     width ) );
-    props.push_back( name_and_value( space + _( "Warmth:" ),
-                                     string_format( "%3d", worn_item.get_warmth() ), width ) );
-    props.push_back( name_and_value( space + string_format( _( "Storage (%s):" ), volume_units_abbr() ),
-                                     format_volume( worn_item.get_storage() ), width ) );
-    return props;
-}
-
-std::vector<std::string> clothing_protection( const item &worn_item, const int width )
+enum sort_armor_columns : int {
+    column_cond,
+    column_encumb,
+    column_storage,
+    column_warmth,
+    column_divider,
+    column_coverage,
+    column_bash,
+    column_cut,
+    column_acid,
+    column_fire,
+    column_env,
+    column_num_entries
+};
+struct sort_armor_col_data {
+    int width;
+    std::string name;
+    std::string description;
+};
+sort_armor_col_data get_col_data( int col_idx )
 {
-    std::vector<std::string> prot;
-    prot.reserve( 4 );
-
-    const std::string space = "  ";
-    prot.push_back( string_format( "<color_c_green>[%s]</color>", _( "Protection" ) ) );
-    prot.push_back( name_and_value( space + _( "Bash:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.bash_resist() ) ), width ) );
-    prot.push_back( name_and_value( space + _( "Cut:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.cut_resist() ) ), width ) );
-    prot.push_back( name_and_value( space + _( "Environmental:" ),
-                                    string_format( "%3d", static_cast<int>( worn_item.get_env_resist() ) ), width ) );
-    return prot;
+    static std::array< sort_armor_col_data, column_num_entries> col_data = { {
+            {
+                3, _( "cond" ), _(
+                    string_format( "clothing conditions:\n\t<color_%s>x</color> - poor fit\n\t<color_%s>f</color> - filthy",
+                                   string_from_color( c_red ), string_from_color( c_cyan ) )
+                )
+            },
+            {8, _( "encum " ), ( "encumberance." )},
+            {7, _( "strge" ), _( "storage." )},
+            {4, _( "wrm" ), _( "warmth." )},
+            {3, "", ""},
+            {3, _( "cvr" ), _( "coverage" )},
+            {4, _( "bsh" ), _( "bash protection." )},
+            {4, _( "cut" ), _( "cut protection." )},
+            {4, _( "acd" ), _( "acid protection." )},
+            {4, _( "fre" ), _( "fire protection." )},
+            {4, _( "env" ), _( "environmental protection." )},
```

Comment:
Even if you don't abbreviate some of the column headers there's still plenty of space to fit them in the screen. Also, as @Leland said, limiting the character count of a string is bad for translation, and I suggest calculating the column width according to the length of the translated text, instead of limiting the maximum characters a column has.